src/parser.js

```diff
@@ -889,13 +898,36 @@ export default class Parser {
         });
       });
 
+      _.forEach(resourceBodyParameters, (param) => {
+        const index = resourceBodyParameters.indexOf(param);
+
+        if (param['x-example']) {
```

Comment:
Agree, but there will be next changes related to `multipart/form-data`.
Let postpone refactoring of this function once done all changes.

There is too `Path` class refactoring proposal as sugest @kylef